waiter/src/waiter/state.clj

```diff
@@ -1303,3 +1308,114 @@
     (t/millis router-syncer-interval-ms)
     #(retrieve-peer-routers discovery router-chan)
     :delay-ms router-syncer-delay-ms))
+
+;; Support for tracking instance scheduling/startup time stats
+
+(defrecord InstanceTracker [service-id known-instance-ids scheduling-instance-timer-contexts starting-instance-trackers])
+
+(defn- new-tracker-val
+  [service-id service-id->service-description-fn]
+  (let [{:strs [min-instances]} (service-id->service-description-fn service-id)
+        ctxs (repeatedly
+               min-instances
+               (fn [] {:schedule [(timers/start (metrics/waiter-timer "service-init-overhead" "schedule-time"))
+                                  (timers/start (metrics/service-timer service-id "service-init-overhead" "schedule-time"))]
+                       :total [(timers/start (metrics/waiter-timer "service-init-overhead" "init-to-healthy-time"))
+                               (timers/start (metrics/service-timer service-id "service-init-overhead" "init-to-healthy-time"))]}))]
```

Comment:
Should we use a map here for the two timer contexts?

Reply:
Same as above.